subsys/net/lib/openthread/platform/entropy.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <string.h>
+#include <drivers/entropy.h>
+#include <logging/log.h>
+
+#include <openthread/platform/entropy.h>
+
+#include "platform-zephyr.h"
+
+LOG_MODULE_REGISTER(net_otPlat_entropy, CONFIG_OPENTHREAD_L2_LOG_LEVEL);
+
+#if defined(CONFIG_ENTROPY_HAS_DRIVER)
+static otError entropy_get(u8_t *buf, u16_t len)
+{
+	/* static to obtain it once in a first call */
+	static struct device *dev;
+	int err;
+
+	if (dev == NULL) {
+		dev = device_get_binding(CONFIG_ENTROPY_NAME);
+		if (!dev) {
+			LOG_ERR("Failed to obtain entropy device");
+			return OT_ERROR_FAILED;
+		}
+	}
+
+	err = entropy_get_entropy(dev, buf, len);
+	if (err != 0) {
+		LOG_ERR("Failed to obtain entropy, err %d", err);
+		return OT_ERROR_FAILED;
+	}
+
+	return OT_ERROR_NONE;
+}
+#else
+static otError entropy_get(u8_t *buf, u16_t len)
```

Comment:
Question, does openthread define whether this needs to be a cryptographic entropy value? Because if the platform doesn't have an entropy driver then it likely isn't cryptographically secure.

Reply:
Their documentation on porting has not caught up with the code changes. But the comments have "This function MUST be implemented using a true random number generator (TRNG)" which implies some level of cryptographic security.

The "MUST" implies a required implementation. I recommend a build error if the platform doesn't support TRNG.

Reply:
No, I'm not aware of such boards, at least not on a Nordic side. I can imagine that we might want to run OT in an emulated environment sometime in the future (like we do with raw 802.15.4 and qemu), but perhaps that not the case for now. I will remove the non-secure implementation.

Reply:
> I can imagine that we might want to run OT in an emulated environment sometime in the future

That's definitely true. But given that we can mock the things on the level of entropy driver then, I'd say that removal of sys_rand32_get-based implementation is indeed not a big loss, and will make our security folks like @d3zd3z happier ;-).